syntax-test_dev.py

```diff
@@ -0,0 +1,271 @@
+        return (None, None)
+
+    lines = []
+    pos = view.sel()[0].begin()
+    first_line = True
+    while pos >= 0:
+        line = get_details_of_test_assertion_line(view, pos)
+        pos = line[2].begin() - 1
+        if line[1]:
+            lines.append(line)
+        elif not first_line or not line[0]:
+            break
+        elif line[0]:
+            lines.append(line)
+        first_line = False
+
+    if len(lines) == 0:
+        return (None, line[2])
+    else:
+        return (lines, line[2])
+
+
+class SyntaxTestEventListener(sublime_plugin.EventListener):
+    """Event Listener for responding to keybinding contexts"""
+
+    def on_query_context(self, view, key, operator, operand, match_all):
+        # all contexts supported will have boolean results, so ignore regex operators
+        if operator not in (sublime.OP_EQUAL, sublime.OP_NOT_EQUAL):
+            return None
+
+        def current_line_is_a_syntax_test():
+            return is_syntax_test_line(view, view.sel()[0].begin(), False)
+
+        def line_above_is_a_syntax_test():
+            line = get_details_of_test_assertion_line(view, view.sel()[0].begin())
+            if line[0] is None:
+                return False
+            else:
+                return is_syntax_test_line(view, line[2].begin() - 1, True)
+
+        def file_contains_syntax_tests():
+            return is_syntax_test_file(view)
+
+        if key not in locals():
+            return None
+        else:
+            result = locals()[key]() == bool(operand)
+            if operator == sublime.OP_NOT_EQUAL:
+                result = not result
+            return result
+
+
+class AlignSyntaxTest(sublime_plugin.TextCommand):
+    """Insert enough spaces so that the cursor will be immediately to the right of the
+    previous line's last syntax test assertion."""
+
+    def run(self, edit):
+        cursor = self.view.sel()[0]
+        line = get_details_of_test_assertion_line(self.view, cursor.begin())
+        if not line[0]:
+            return
+
+        # find the last test assertion column on the previous line
+        line = get_details_of_test_assertion_line(self.view, line[2].begin() - 1)
+        if line[2] is not None:
+            self.view.insert(edit, cursor.end(), ' ' * (
+                line[1][1] - self.view.rowcol(cursor.begin())[1]
+            ))
+        self.view.run_command('suggest_syntax_test')
+
+
+class SuggestSyntaxTest(sublime_plugin.TextCommand):
+    """Intelligently suggest where the syntax test assertions should be placed,
+    based on the scopes on the line being tested, and where they change."""
+
+    def run(self, edit, **args):
+        """Available parameters:
+        edit (sublime.Edit)
+            The edit parameter from TextCommand.
+        character (str) = '^'
+            The character to insert when suggesting where the test assertions should go.
+        """
+
+        view = self.view
+        char = args.get('character', '^')
```

Comment:
I did this for consistency when typing the `-` after `<`, i.e. to test the scope at the comment marker